designs/auto-certs-renewal.md

```diff
@@ -0,0 +1,132 @@
+
+### External ETCD node certificates
+
+* etcd node’s certificates are stored in /etc/etcd/pki (/var/lib/etcd/pki in bottlerocket)
+* each certificate comes in pair: key and crt
+* all etcd nodes contain the same key pair names no matter if it’s the first node (not in term of contents)
+
+list of all the certificate files:
+
+1. apiserver-etcd-client pair: generated by etcdadm,  apiserver can use it to access etcd, **but it’s not used**
+2. ca pair: the only pair which is not generated by etcdadm, it comes from cloud-init script.
+3. etcdctl-etcd-client pair: generated by etcdadm during `etcdadm join <endpoint>`
+4. peer pair: generated by etcdadm `etcdadm join <endpoint>`
+5. server pair: generated by etcdadm `etcdadm join <endpoint>`
+
+### CP node certificates
+
+* all certificates are located in /etc/kubernetes/pki  (/var/lib/kubeadm/pki in bottlerocket)
+* all cp nodes contains the same key pair names
+
+list of all the certificate files:
+
+1. apiserver-etcd-client pair: from cloud-init script, but the source is etcdadm controller 
+2. ca pair: from cloud-init script
+3. front-proxy-ca pair: from cloud-init script
+4. sa pair: from cloud-init script
+5. etcd/ca.crt: from cloud-init script
+6. apiserver-kubelet-client pair: from kubeadm
+7. apiserver pair: from kubeadm
+8. front-proxy-client pair: from kubeadm
+
+The above certificate files are described in the official doc: https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates
+
+### Worker node certificates
+
+1. just ca.crt in /etc/kubernetes/pki, which should be generated after `kubeadm join`
+
+## Goal
+
+1. Auto renew all the above mentioned certificates except etcd CA and kube CA, which last for 10 years.
+
+### Observation
+
+1. We can renew all etcd’s certs by simply **recreating the external etcd machine**. If doing manually, we just need to delete the auto generated certs and run `sudo etcdadm join phase certificates`
+2. We can renew all cp’s certs besides apiserver-etcd-client by **recreating the cp machine.** As we can see from the following diagram that the apiserver-etcd-client is generated by Etcdadm Controller, so this cert cannot be renewed by just recreating the cp machine.
+3. No need to do auto renew certs in worker nodes.
+
+## Solution
+
+Etcd can be deployed in [two topologies.](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/ha-topology/) Certs renew mechanism is slightly different in those two scenarios.
+
+### Stacked etcd Nodes
+
+In Stacked etcd nodes topology, etcd is running inside the cp node. 
+
+We can renew all certificates by just redeploying cp nodes
+
+1. Expose [rolloutBefore.certificatesExpiryDays](https://cluster-api.sigs.k8s.io/tasks/certs/auto-rotate-certificates-in-kcp.html) field from KubeadmControlPlane in EKSA cluster spec. Add pass the field to KubeadmControlPlane
```

Comment:
nit: we are not exposing `certificatesExpiryDays` directly in out spec. So maybe "control rolloutBefore.certificatesExpiryDays from EKS-A, setting a fixed value (7 days) when auto-renew certificates is enabled". Or something like that.